src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -1784,16 +1754,11 @@ inline fun processWithShadowingAndUpdateScope(
 inline fun processWithShadowing(
     prevScope: Map<String, Set<Namespace>>,
     ns: Set<Namespace>,
-    originalProcessor: RsResolveProcessor,
+    processor: RsResolveProcessor,
     f: (RsResolveProcessor) -> Boolean
 ): Boolean {
-    val processor = createProcessor(originalProcessor.names) { e ->
-        val prevNs = prevScope[e.name]
-        if (e.name == "_" || prevNs == null) return@createProcessor originalProcessor(e)
-        val newNs = (e.element as? RsNamedElement)?.namespaces
-        (newNs == null || ns.intersects(newNs.minus(prevNs))) && originalProcessor(e)
-    }
-    return f(processor)
+    val shadowingProcessor = processor.wrapWithShadowingProcessor(prevScope, ns)
```

Comment:
Is it needed to introduce `wrapWithShadowingProcessor` and `ShadowingProcessor` given that they are used only in one place in `processWithShadowing`?

Reply:
Why not? The most reason is having good processor names instead of anonymous lambdas in the profiles. So we should think not about how often the function is used, but how often it is actually executed during name resolution (spoiler: all the time =D)